### PR TITLE
Implemented functionality to add,show and remove Favourite Questions for a user attempting the exam

### PR DESCRIPTION
**Pull request by Shiwang0-0**
I have added the functionality to add a question to the favourites after the user starts attempting the exam.
- The user can add a specific question to the favourites.
- The user can view the favourites later.
- The user can remove a specific question from the favourites later.

Controllers performing POST request to add and remove favourite question is added, a GET method is also added to fetch all the favourites which remains in the collection.

Schema for Favourite question collection is added which stores the question name, user id (to determine favourites of a specific user) and question's option.

### CODE REVIEW

**Comment by Shiwang0-0** [comment]
Functionaltiy to add,remove,show the favourite questions for a user attempting the quiz.

File: backend/src/controllers/user.ts [hunk 1]
```diff
@@ -210,6 +211,60 @@ const deactivateUserCallback: RequestHandler = async (req, res, next) => {
...
+  try {
+    const user = await User.findById(userId);
+    if (!user) {
+      const err = new ProjectError("User does not exist");
+      err.statusCode = 401;
+      throw err;
+    }
+    
+    const favQues = new Fav({ question, options,userId });    const favQuestion = await favQues.save();
+    resp = { status: "success", message: "Question added to Favourites!", data: {} };
```
Comment (vermaaakarsh) [inline backend/src/controllers/user.ts]: what is the user of this favQuestion variable?

File: backend/src/controllers/user.ts [hunk 2]
```diff
@@ -210,6 +211,60 @@ const deactivateUserCallback: RequestHandler = async (req, res, next) => {
...
+  catch (error) {
+    next(error);
+  }
+};
+
+const showFavQues: RequestHandler = async (req, res, next) => {
+  const userId = req.userId;
+  let resp: ReturnResponse;
+  try {
+    const fav = await Fav.find({userId});
```
Comment (vermaaakarsh) [inline backend/src/controllers/user.ts]: this should be named favQuestion, follow a meaningful naming habit.

File: backend/src/routes/user.ts [hunk 3]
```diff
@@ -32,6 +35,15 @@ router.patch("/deactivate", isAuthenticated, deactivateUser);
...
 router.get("/deactivate/:token", deactivateUserCallback)
 
+//Post /user/addfavques
+router.post("/addfavques", isAuthenticated,addFavQues );
+
+//Get /user/favques
+router.get("/favques", isAuthenticated,showFavQues );
+
+//Post /user/removefavques
+router.post("/removefavques", isAuthenticated,removeFavQues );
```
Comment (vermaaakarsh) [inline backend/src/routes/user.ts]: why this is not delete method? And where are the validations?

**Review by vermaaakarsh — CHANGES_REQUESTED** [review]

**Comment by vermaaakarsh** [comment]
Your title is the merge message that will be displayed in the commits, make it more descriptive and follow the pattern

File: backend/src/controllers/user.ts (hunk 2 above)
Comment (Shiwang0-0) [reply backend/src/controllers/user.ts]: noted

File: backend/src/routes/user.ts (hunk 3 above)
Comment (Shiwang0-0) [reply backend/src/routes/user.ts]: noted,made the changes.

**Comment by Shiwang0-0** [comment]
Done with the changes recommened @vermaaakarsh sir

**Pull request merged**